src/eureka/S5_lightcurve_fitting/models/BatmanModels.py

```diff
@@ -45,9 +45,24 @@ def __init__(self, **kwargs):
         self.paramtitles = kwargs.get('paramtitles')
 
         # Store the ld_profile
-        ld_func = ld_profile(self.parameters.limb_dark.value)
+        self.ld_from_S4 = kwargs.get('ld_from_S4')
+        ld_func = ld_profile(self.parameters.limb_dark.value, 
+                             use_gen_ld=self.ld_from_S4)
         len_params = len(inspect.signature(ld_func).parameters)
         self.coeffs = ['u{}'.format(n) for n in range(len_params)[1:]]
+        
+        # Replace fixed u parameters with generated limb-darkening values
+        if self.ld_from_S4 is not None:
+            self.ld_S4_array = kwargs.get('ld_coeffs')[len_params-2]
```

Comment:
I'm not sure what went wrong with the automated tests, but when I run the tests locally I get the following error at this line for the MIRI and NIRCam tests (didn't try the other instruments)
```
        # Replace fixed u parameters with generated limb-darkening values
        if self.ld_from_S4 is not None:
>           self.ld_S4_array = kwargs.get('ld_coeffs')[len_params-2]
E           TypeError: 'NoneType' object is not subscriptable

/Users/tjbell1/Research/JWST/ERS/Eureka/src/eureka/S5_lightcurve_fitting/models/BatmanModels.py:56: TypeError
```